src/integrations/Http2Integration.ts

```diff
@@ -0,0 +1,275 @@
+
+                    span = tracer._startSpan(operationName, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.HTTP2,
+                        disableActiveStart: true,
+                    });
+
+                    if (!config.httpTraceInjectionDisabled) {
+                        const tempHeaders = headers ? headers : {};
+                        tracer.inject(span.spanContext, opentracing.FORMAT_TEXT_MAP, tempHeaders);
+                        tempHeaders[TriggerHeaderTags.RESOURCE_NAME] = operationName;
+                        headers = tempHeaders;
+                    }
+
+                    span.addTags({
+                        [SpanTags.OPERATION_TYPE]: method,
+                        [SpanTags.SPAN_TYPE]: SpanTypes.HTTP2,
+                        [Http2Tags.HTTP2_METHOD]: method,
+                        [Http2Tags.HTTP2_HOST]: host,
+                        [Http2Tags.HTTP2_PATH]: path,
+                        [Http2Tags.HTTP2_URL]: fullURL,
+                        [Http2Tags.QUERY_PARAMS]: queryParams,
+                        [SpanTags.TRACE_LINKS]: [span.spanContext.spanId],
+                        [SpanTags.TOPOLOGY_VERTEX]: true,
+                    });
+
+                    const me = this;
+
+                    span._initialized();
+
+                    const clientRequest = request.apply(this, [headers, options]);
+
+                    const chunks: any = [];
+                    let responseHeaders: any;
+
+                    clientRequest.once('data', (chunk: any) => {
+                        if (!chunk) {
+                            return;
+                        }
+
+                        const totalSize = chunks.reduce((total: any, item: any) => item.length + total, 0);
+                        if (totalSize + chunk.length <= 10 * 1024) {
+                            chunks.push(typeof chunk === 'string' ? Buffer.from(chunk) : chunk);
+                        }
+                    });
+
+                    clientRequest.once('response', (res: any) => {
+                        responseHeaders = HTTPUtils.extractHeaders(res);
+
+                        ThundraLogger.debug(`<HTTP2Integration> On response of HTTP2 span with name ${operationName}`);
+                    });
+
+                    clientRequest.once('end', () => {
+
+                        let payload;
+                        try {
+                            payload = JSON.parse(chunks);
+                        } catch (error) {
+                            ThundraLogger.debug('<HTTP2Integration> Response is not valid JSON:', payload);
+                            payload = chunks.toString();
+                        }
+
+                        if (span) {
+                            const statusCode = responseHeaders[':status'] || 200;
```

Comment:
Can you also apply the following logic from `HttpIntegration` (maybe move to `HTTPUtils`?):
```js
if ('x-amzn-requestid' in res.headers) {
    span._setClassName(ClassNames.APIGATEWAY);
}
if (TriggerHeaderTags.RESOURCE_NAME in res.headers) {
    const resourceName: string = res.headers[TriggerHeaderTags.RESOURCE_NAME];
    span._setOperationName(resourceName);
}
```